clojure/fira_code/checks.clj

```diff
@@ -4,20 +4,33 @@
    [fira-code.coll :as coll]
    [fira-code.glyphs :as glyphs]))
 
+(def expected-widths #{"0" 0 1200})
+
 (defn width-ok? [w]
-  (#{"0" 0 1200} w))
+  (expected-widths w))
 
 (defn widths [font]
-  (doseq [g     (:glyphs font)
-          :when (not= "0" (:export g))
-          :let  [[w & _ :as ws] (mapv :width (:layers g))]]
-    (when-not (apply = ws) 
-      (println (str "WARN glyph '" (:glyphname g) "' has different widths=" (pr-str ws))))
-    (when-not (width-ok? w)
-      (println (str "WARN glyph '" (:glyphname g) "' has unexpected width=" (pr-str w)))))
-  font)
+  (let [warnings (for [g (:glyphs font)
+                       :when (not= "0" (:export g))
+                       :let [ws (mapv :width (:layers g))]]
+                   (cond
+                     (not (apply = ws))
+                     (str "Glyph '" (:glyphname g) "' has different widths: " (pr-str ws))
+
+                     (not (width-ok? (first ws)))
+                     (str "Glyph '" (:glyphname g) "' has unexpected width: " (pr-str (first ws)))
+
+                     :else nil))]
+    (remove nil warnings)))
```

Comment:
Here you just delay printing warnings, adding extra data structure that’s passed between widths and its call site. That’s one more thing to keep in mind, which makes it harder to follow